IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -421,16 +431,18 @@ - (BOOL)saveAppMetadata:(nonnull MSIDAppMetadataCacheItem *)metadata
     
     MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, context, @"Saving app's metadata %@", MSID_PII_LOG_MASKABLE(metadata));
     
-    MSIDAppMetadataCacheKey *key = [[MSIDAppMetadataCacheKey alloc] initWithClientId:metadata.clientId
-                                                                         environment:metadata.environment
-                                                                            familyId:metadata.familyId
-                                                                         generalType:MSIDAppMetadataType];
-
-    return [_dataSource saveAppMetadata:metadata
-                                    key:key
-                             serializer:_serializer
-                                context:context
-                                  error:error];
+    MSIDAppMetadataCacheKey *key = (MSIDAppMetadataCacheKey *)[metadata generateCacheKey];
+    
+    if ([key isKindOfClass:[MSIDAppMetadataCacheKey class]])
+    {
+        return [_dataSource saveAppMetadata:metadata
```

Comment:
doesn't saveAppMetadata accept just a generic cachekey? Wouldn't casting and class checking be unnecessary then?

Reply:
The casting was necessary for api's which try to set subclass specific properties (MSIDDefaultAccountCacheKey) on the key which are not declared on the base class MSIDCacheKey. Without the cast, we cannot access those subclass specific properties. 

For instance in saveAccount we try to set key.username and in removeCredential we try to set key.appKey. I added casting in all api's for consistency. But agree, that it is not required everywhere. 

Updated code to only add casting in the api's mentioned above.